src/data_race.rs

```diff
@@ -192,6 +192,25 @@ struct AtomicMemoryCellClocks {
     sync_vector: VClock,
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+enum WriteType {
+    /// Allocate memory.
+    Allocate,
+    /// Standard unsynchronized write.
+    Write,
+    /// Deallocate memory
+    Deallocate,
+}
```

Comment:
To make sure that I understand this correctly... if it's the dealloc happening first, there won't be a race reported but an out-of-bounds access, right? It's only when the read is first, and then a dealloc follows, but it is not synchronized, *then* a race is reported? And same for allocations? I think this would be worth documenting somewhere.

Also, this is a further deviation from the paper, right? So please add it to the list of "differences from the paper" at the top of this file.